desktop/mgtk.s

```diff
@@ -1171,110 +1296,132 @@ fill_mode_table_onechar:
 ;;; ============================================================
 ;;; SetPenMode
 
-SetPenModeImpl:
+.proc SetPenModeImpl
         lda     current_penmode
         ldx     #0
         cmp     #4
         bcc     :+
         ldx     #$7F
 :       stx     fill_eor_mask
         rts
+.endproc
 
         ;; Called from PaintRect, DrawText, etc to configure
         ;; fill routines from mode.
-set_up_fill_mode:
+
+.proc set_up_fill_mode
+        x1      := $92
+        x2      := $96
+
+        x1_char := $86
+        x2_char := $82
+
         lda     x_offset
         clc
-        adc     $96
-        sta     $96
-        lda     $F8
-        adc     $96+1
-        sta     $96+1
-        lda     $F8+1
+        adc     x2
```

Comment:
aside: the lda/clc/adc/sta/lda/adc/sta pattern can use the add16 macro. My refactor tool probably missed it here because x_offset wasn't used consistently. It should catch it now.